optaplanner-core/src/main/java/org/optaplanner/core/api/score/holder/AbstractScoreHolder.java

```diff
@@ -224,4 +229,18 @@ public final void run() {
             }
         }
     }
+
+    /**
+     * Used for Drools-based constraint streams (CS-D) to convert justification list to the same format as the one used
+     * by Bavet constraint streams.
+     * This is necessary because CS-D uses some advanced Drools constructions leveraging various metadata objects.
+     * Had we not called this adapter, these metadata objects would have been present in the justification list,
+     * defeating its purpose.
+     * Code outside of CS-D has no need to call this method.
+     *
+     * @param justificationListAdapter function to clean up the justification list
+     */
+    public void setJustificationListAdapter(BiFunction<List<Object>, Rule, List<Object>> justificationListAdapter) {
+        this.justificationListAdapter = justificationListAdapter;
+    }
```

Comment:
I don't really like that I had to put this on the public API.
I considered to put it in the constructor, but that would require me to update every single `ScoreHolder` impl.
I'm listening for other ideas.

Reply:
Yea :(
Let's discuss on bluejeans.

Reply:
We keep the method. We rename it to *Converter.